### PR TITLE
Refresh project pane

### PR DESCRIPTION
**Pull request by nashousni**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
no

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
Project pane Tree items are not always refreshed correctly during modifications.

**What is the new behavior (if this is a feature change)?**
When renaming an object or correcting an incomplete object, the corresponding tree item is automatically refreshed in order to update the modification time , and correct incomplete nodes

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
no breaking change

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java [hunk 1]
```diff
@@ -161,6 +167,35 @@ public void contentChanged() {
...
+                    .findFirst();
+            projectFileItem.ifPresent(treeItem -> refresh(treeItem.getParent()));
+        }
+    }
+
+    private void findAllVisibleTreeItems(TreeItem<Object> folderTreeItem, List<TreeItem<Object>> allVisibleTreeItems) {
+        ObservableList<TreeItem<Object>> children = folderTreeItem.getChildren();
+        allVisibleTreeItems.addAll(children);
+        children.stream()
+                .filter(item -> ((ProjectNode) item.getValue()).isFolder() && item.isExpanded())
```
Comment (pl-buiquang) [inline gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java]: must be done on all nodes even those who ain't visible

File: gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java [hunk 2]
```diff
@@ -161,6 +167,35 @@ public void contentChanged() {
         };
     }
 
+    private void handleEvent(NodeEvent nodeEvent) {
+        ProjectFile projectFile = project.getFileSystem().findProjectFile(nodeEvent.getId(), ProjectFile.class);
+        if (NodeDataUpdated.TYPENAME.equals(nodeEvent.getType()) || DependencyAdded.TYPENAME.equals(nodeEvent.getType())) {
```
Comment (pl-buiquang) [inline gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java]: check if projectFile's project is the same as this ProjectPane's
we don't want to try refreshing a node that for sure doesn't exist here

Comment (nashousni) [reply gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java]: resolved

File: gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java (hunk 1 above)
Comment (nashousni) [reply gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java]: resolved

**Pull request merged**